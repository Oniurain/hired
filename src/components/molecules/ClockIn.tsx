import * as React from 'react'
import { observer } from 'mobx-react'
import styled from 'styled-components'

import { Button } from '../atoms/'
import { standardHPadding } from '../../styles/style-utils'

interface ClockInProps {
  className?: string
  running: boolean,
  startEntry: Function,
  stopEntry: Function
}

const ClockInButton = styled(Button)`
  margin: 0 ${standardHPadding};
  width: calc(100vw - 2*${standardHPadding});
`

@observer
class ClockIn extends React.Component<ClockInProps, {}> {
  constructor(props: ClockInProps) {
    super(props)

    this.startEntry = this.startEntry.bind(this)
    this.stopEntry = this.stopEntry.bind(this)
  }

  render() {
    return(
      <ClockInButton
        success={(!this.props.running)}
        error={(!!this.props.running)}
        onClick={(!this.props.running) ? this.startEntry : this.stopEntry}
      >
        {`Clock ${(this.props.running ? 'Out' : 'In')}`}
      </ClockInButton>
    )
  }

  startEntry() {
    this.props.startEntry()
  }

  stopEntry() {
    this.props.stopEntry()
  }
}

export default ClockIn
