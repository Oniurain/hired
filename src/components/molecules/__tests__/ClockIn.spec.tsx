import * as React from 'react'
import { mount } from 'enzyme'

import ClockIn from '../ClockIn'

describe('ClockIn', () => {
  it('should render in idle-state', () => {
    const subject = mount(
      <ClockIn
        running={false}
        startEntry={() => {}}
        stopEntry={() => {}}
      />
    )

    expect(subject).toMatchSnapshot()
  })

  it('should render in running-state', () => {
    const subject = mount(
      <ClockIn
        running={true}
        startEntry={() => {}}
        stopEntry={() => {}}
      />
    )

    expect(subject).toMatchSnapshot()
  })
})
