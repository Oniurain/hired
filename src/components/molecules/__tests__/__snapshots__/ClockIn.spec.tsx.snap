// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClockIn should render in idle-state 1`] = `
<ClockIn
  running={false}
  startEntry={[Function]}
  stopEntry={[Function]}
>
  <styled.div>
    <div
      className="frIlRZ"
    >
      <styled.button
        data-t-target="ClockInButton"
        error={false}
        onClick={[Function]}
        success={true}
      >
        <button
          className="hvWpBV"
          data-t-target="ClockInButton"
          onClick={[Function]}
        >
          Clock In
        </button>
      </styled.button>
    </div>
  </styled.div>
</ClockIn>
`;

exports[`ClockIn should render in running-state 1`] = `
<ClockIn
  running={true}
  startEntry={[Function]}
  stopEntry={[Function]}
>
  <styled.div>
    <div
      className="frIlRZ"
    >
      <styled.button
        data-t-target="ClockInButton"
        error={true}
        onClick={[Function]}
        success={false}
      >
        <button
          className="etMWMq"
          data-t-target="ClockInButton"
          onClick={[Function]}
        >
          Clock Out
        </button>
      </styled.button>
    </div>
  </styled.div>
</ClockIn>
`;
